docs/maui/layouts/StateContainer.md

```diff
@@ -145,6 +145,34 @@ public partial class StateContainerViewModel
 }
 ```
 
+By default `StateContainer` changes state without animation. To add a custom animation you can use the `ChangeStateWithAnimation` method:
+
+```csharp
+async Task ChangeStateWithCustomAnimation()
+{
+    var targetState = "TargetState";
+    var currentState = StateContainer.GetCurrentState(MyBindableObject);
+    if (currentState == targetState)
+    {
+        await StateContainer.ChangeStateWithAnimation(
+            MyBindableObject,
+            null,
+            (element, token) => element.ScaleTo(0, 100, Easing.SpringIn).WaitAsync(token),
+            (element, token) => element.ScaleTo(1, 250, Easing.SpringOut).WaitAsync(token),
+            CancellationToken.None);
+    }
+    else
+    {
+        await StateContainer.ChangeStateWithAnimation(
+            MyBindableObject,
+            targetState,
+            (element, token) => element.ScaleTo(0, 100, Easing.SpringIn).WaitAsync(token),
+            (element, token) => element.ScaleTo(1, 250, Easing.SpringOut).WaitAsync(token),
+            CancellationToken.None);
+    }
```

Comment:
Is there any value in making these animations different to make the distinction clearer?

Reply:
I explicitly used the same animation to not make people confused why we use different animation for different states. Do you think it is better to use different animations?

Reply:
> Do you think it is better to use different animations 

This is the same implementation leveraged from the Sample App: https://github.com/CommunityToolkit/Maui/blob/49aeb96ea34ec2aab0ec224847f41e2772c22d6a/samples/CommunityToolkit.Maui.Sample/Pages/Layouts/StateContainerPage.xaml.cs#L30-L47

As an app developer, I think it's important for the user to experience the same animation every time they click a button. I think that seeing a different animation after clicking the same button would be jarring unless the app is explicitly designed to do so. 

Here's a GIF of how it looks. Maybe we should add this GIF to the docs to provide additional context to the custom animation? 
![Simulator Screen Recording - iPhone 14 Pro - 2023-02-28 at 12 35 45](https://user-images.githubusercontent.com/13558917/221972986-aca4a387-02f2-4e76-8749-dae54aeffe3a.gif)

Reply:
it at least changes with animation. that demonstrates how it works. and this is the only 1 image we have. I don't know if there is a limitation for the image size, but for video, we should not extand more than 10 MB.

Reply:
I completely agree we want to show a common experience for the user. I guess I wanted some context as a reader to the differences between the calls in the of statement. As far as I can see it's just whether we pass `null` or `targetState` in. What does this do? Can we make that clearer to the reader?